src/Search/Service/ScopableRequestTypeConfiguration.php

```diff
@@ -36,7 +36,8 @@ public function getConfig(array $scopedConfig, ?LocalizedCatalog $localizedCatal
 
         $defaultConfig = $scopedConfig['global']['request_types']['generic'];
         $defaultLocalizedConfig = $scopedConfig[$localizedCatalogCode]['request_types']['generic'] ?? [];
-        $defaultConfig = array_replace_recursive($defaultConfig, $defaultLocalizedConfig);
+        $defaultRequestTypeConfig = $scopedConfig['global']['request_types'][$requestType] ?? [];
+        $defaultConfig = array_replace_recursive($defaultConfig, $defaultLocalizedConfig, $defaultRequestTypeConfig);
```

Comment:
you should  update function description to include this new tree config available.